packages/node-experimental/src/sdk/api.ts

```diff
@@ -0,0 +1,225 @@
+// PUBLIC APIS
+
+import { context } from '@opentelemetry/api';
+import { DEFAULT_ENVIRONMENT, closeSession, makeSession, updateSession } from '@sentry/core';
+import type {
+  Breadcrumb,
+  BreadcrumbHint,
+  CaptureContext,
+  Client,
+  Event,
+  EventHint,
+  EventProcessor,
+  Extra,
+  Extras,
+  Primitive,
+  Session,
+  Severity,
+  SeverityLevel,
+  User,
+} from '@sentry/types';
+import { GLOBAL_OBJ, consoleSandbox, dateTimestampInSeconds } from '@sentry/utils';
+import { getScopesFromContext, setScopesOnContext } from '../utils/contextData';
+
+import type { ExclusiveEventHintOrCaptureContext } from '../utils/prepareEvent';
+import { parseEventHintOrCaptureContext } from '../utils/prepareEvent';
+import type { Scope } from './scope';
+import { getClient, getCurrentScope, getGlobalScope, getIsolationScope } from './scope';
+
+export { getCurrentScope, getGlobalScope, getIsolationScope, getClient };
+export { setCurrentScope, setIsolationScope } from './scope';
+
+/**
+ * Fork a scope from the current scope, and make it the current scope in the given callback
+ */
+export function withScope<T>(callback: (scope: Scope) => T): T {
+  return context.with(context.active(), () => callback(getCurrentScope()));
+}
+
+/**
+ * Fork a scope from the current scope, and make it the current scope in the given callback.
+ * Additionally, also setup a new isolation scope for the callback.
+ */
+export function withIsolationScope<T>(callback: (scope: Scope, isolationScope: Scope) => T): T {
```

Comment:
hmm. Just thinking about things naively I find it odd that withIsolationScope would create more than just an additional scope that happens to be an isolation scope. The way I initially read this function would have been that `scope` always is the same as `isolationScope`. Is that not the case? I would expect the `currentScope` inside `withIsolationScope` to be the isolation scope.

Reply:
no, the isolation scope is never a current scope - like the global scope, it is completely separate. otherwise, inheritance of scopes would be messed up - think this:

```js
withIsolationScope(scope => {
  // scope is both current & isolation scope
  scope.setTag('xx', 'yy');
  withScope(innerScope => {
    // innerScope inherits from the previous current scope, 
    // which would now be the isolation scope, which is bad
  });
});
```

Reply:
Why is it bad to fork from the isolation scope?

Reply:
it's bad if that happens implicitly, without the user knowing this. Keep in mind that in 99% of cases users will never call `withIsolationScope` manually but the isolation scope will be set under the hood, and users should not accidentally mutate it by accessing `getCurrentScope()` and putting something on it.

I've updated the function syntax to _only_ pass the isolation scope for `withIsolationScope`.

You can refer to this graphic which hopefully explains what inherits from what where: https://github.com/getsentry/rfcs/blob/c4b1fb4fd2915246760156285393340af093d591/text/0122-images/scope-inheritance.png